rules/aws_s3_bucket_name.go

```diff
@@ -100,12 +177,18 @@ func (r *AwsS3BucketNameRule) Check(runner tflint.Runner) error {
 				}
 			}
 
-			if len(name) < bucketNameMinLength || len(name) > bucketNameMaxLength {
-				runner.EmitIssue(
-					r,
-					fmt.Sprintf("Bucket name %q must be between %d and %d characters", name, bucketNameMinLength, bucketNameMaxLength),
-					attribute.Expr.Range(),
-				)
+			for _, rule := range rules {
+				regex, err := regexp.MatchString(rule["regx"], name)
+				if err != nil {
+					return fmt.Errorf("invalid regex (%s): %w", rule["regx"], err)
+				}
+				if regex {
+					runner.EmitIssue(
+						r,
+						fmt.Sprintf(`Bucket name %q does not match regex %q: %q`, name, rule["regx"], rule["desc"]),
```

Comment:
Reverse the order to description then regex. The description is for humans, whereas the regex is a machine-interpreted language, offered as an additional diagnostic aid. In some cases the regexes are quite long and push the description far to the right. A rough idea:

```go
fmt.Sprintf("%s (name: %q, regex: %q)", rule.Description, name, rule.Regexp.String())
```